libr/main/rasm2.c

```diff
@@ -708,6 +708,28 @@ static void __load_plugins(RAsmState *as) {
 	free (path);
 }
 
+static inline char *io_slurp(const char *file, size_t *len) {
+	RIODesc *des;
+	ut8 *ret = NULL;
+	RIO *io = r_io_new ();
+	if (io) {
+		des = r_io_open_nomap (io, file, R_PERM_R, 0);
+		if (des) {
+			ut64 size = r_io_desc_size (des);
```

Comment:
size can be really large. i dont think we want to assemble 3GB files for example. Maybe better if rasm could use RBuffer, which can be backed by io, this way it wont need to slurp the data

Reply:
I added a check for `size >= ST32_MAX` for now. I like the idea of using `RBuffer`. It would likely simplify the code.

I would have to get deeper into the code to get that working though and I don't have a lot of free time atm. If you like the update you can merge, or wait until I make it into something better.